examples/nextjs/src/components/my-page.js

```diff
@@ -49,13 +50,20 @@ export function MyPage({ pageAsset, nav }) {
      * If not using experiments:
      * - Replace the below line with `const DotLayoutComponent = DotcmsLayout;`
      * - Remove DotExperimentsProvider from the return statement.
+     * 
+     * We use a useMemo to prevent the component from being recreated on every render
+     * if it doesn't change.
      */
-    const DotLayoutComponent = experimentConfig.apiKey
-        ? withExperiments(DotcmsLayout, {
-              ...experimentConfig,
-              redirectFn: replace,
-          })
-        : DotcmsLayout;
+    const DotLayoutComponent = useMemo(
+        () =>
+            experimentConfig.apiKey
+                ? withExperiments(DotcmsLayout, {
+                      ...experimentConfig,
+                      redirectFn: replace,
+                  })
+                : DotcmsLayout,
+        [replace],
+    );
```

Comment:
Can we use the `useMemo` inside the `withExperiments` so the customer don't have to remember to do this?

So, the problem is something we are causing in our library and this solution relies on the customer... we should not put this in his lap.

Reply:
After trying it, I would think not
The problem is that the usePageAsset hook is going to re-render the component resulting from withExperiments

When the DotCmsLayout component is rendered, there is a useEffect that depends on the current path to trigger the usePageAsset hook.

I think, since withExperiments is a [HOC](https://legacy.reactjs.org/docs/higher-order-components.html), the resulting component will always have a different reference and the useEffect that depends on the current path will always be triggered

One other option is to create a "flat" component like `<DotcmsLayoutWithExperiment>` and put a conditional in the same template

Reply:
Other option if change the flow on initEditor, because that trigger the render with usePageAsset hook
I'm going to try to change it

Reply:
We gotta figure something out... because this solution feels like a hack that we're giving the reponsability of the customer to do.

Reply:
Later than we talk about the GQL issue, we need to mantain that first call from client to UVE
So i created a hook to memoize the object reference
I just updated the PR